docs/DOCUMENTATION.md

```diff
@@ -0,0 +1,872 @@
+  c. There is a deviation of more than W% from the current price on-chain. Once again, only done if the oracle has not started the previous round himself.
+7. Keep the following information in the state file for each feed:
+  a. The CL job's external id for that feed
+  b. The name of the feed
+  c. The last request id
+  d. The last reported round on chain
+  e. The timestamp of the last request made to the CL node
+  f. The last aggregated price on chain
+  g. The request id of the last result received from the CL node
+  h. The details of the latest on-chain round including the round id, the timestamp when it was started, by whom it was started and whether the oracle made a submission for that round
+
+#### Monitoring
+
+The monitoring script is needed to:
+1. Monitor the actual price and that it is being updated
+2. Monitor node operators submissions to ensure they are submitting values within accepatable thresholds and that they are submitting to rounds and not missing them
+3. Monitor node operators' balances to ensure they have enough balance for transaction fees to push prices on chain
+4. Monitor the rate at which rounds are being created
+
+The monitoring script should contain the following functionalities:
+
+1. A state file which is updated so that whenever the monitoring script is restarted it is able to continue from where it was stopped.
+2. Monitor multiple oracles at once
+3. An endpoint to expose prometheus metrics 
+4. Expose the following metrics
+  a. The latest submitted value on chain by an oracle for a feed
+  b. The timestamp in which an oracle made an on-chain submission for a feed 
+  c. The last round for which an oracle made an on-chain submission for a feed
+  d. The deviation of an oracle's submitted price from the latest aggregated value on-chain for a feed
+  e. The oracle balance 
+  f. The actual price on-chain
+4. Obtain an oracle's invitation IDs for feeds from wallets
+5. Query the latest prices and round submissions of oracles every X seconds and update the metrics
+6. Have an efficient way of polling only the latest price pushes so be able to monitor oracles efficiently when the number of offers used start to increase
+
+
+## File structure
+
+In this section, one is able to find a description of the file structure and the contents of files
+
+#### scripts
+
+The scripts directory contains scripts which are used to for deployments. 
+
+The following are the different scripts which can be found in this directory
+
+1. accept-oracle-invitation.sh - This script can be used to accept an oracle invitation. This script takes in 3 parameters, the wallet name, the brand in and brand out. An example of a command to run this is ```./accept-oracle-invitation.sh $WALLET_NAME $BRAND_IN $BRAND_OUT```
```

Comment:
What is brand in and brand out?

Reply:
It's not really an example if you put variables, show concrete examples

Reply:
They are base and quote.
For example, in "ATOM-USD", ATOM would be 'brand in' and USD would be 'brand out'

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/8a77991926f6a3d1d36d22028f54ca0f64e17527